File: code/day1.py
# process
with open('inputs/1.txt') as f:
    lines = f.readlines()
    nums = [int(line.strip()) for line in lines]

# part 1
part1count = 0
for index, num in enumerate(nums):
    number = num
    if index < len(nums) - 1:
      nextNum = nums[index+1]
    if nextNum > num:
      part1count += 1
print('part 1:', part1count)

# Your puzzle answer was 1451.

# part 2

def getWindow(idx):
  return nums[idx] + nums[idx+1] + nums[idx+2]

part2count = 0
for index, num in enumerate(nums):
    if index == len(nums) - 3:
      break
    currentWindow = getWindow(index)
    nextWindow = getWindow(index+1)
    if nextWindow  > currentWindow:
      part2count += 1
print('part 2:', part2count)

# Your puzzle answer was 1395

File: code/day1.js
// read file
const fs = require("fs").promises;
async function getArray() {
  const data = await fs.readFile("inputs/1.txt", "utf8");
  const numbers = data.split("\n");
  return numbers.map((num) => Number(num));
}

function question1(numbers) {
  return numbers.reduce((total, currentNum, idx, arr) => {
    if (idx === numbers.length - 1) {
      return total;
    }
    const nextNum = arr[idx + 1];
    if (nextNum > currentNum) {
      total++;
    }
    return total;
  }, 0);
}

function question2(numbers) {
  return numbers.reduce((total, currentNum, idx, arr) => {
    if (idx === numbers.length - 3) {
      return total;
    }
    const currentWindow = currentNum + arr[idx + 1] + arr[idx + 2];
    const nextWindow = arr[idx + 1] + arr[idx + 2] + arr[idx + 3];
    if (nextWindow > currentWindow) {
      total++;
    }
    return total;
  }, 0);
}

(async function main() {
  const data = await getArray();
  console.log(question1(data)); // 1451
  console.log(question2(data)); // 1395
})();
